### DISCUSSION TITLE
Tower Defense - SRS

### DISCUSSION THREAD

**Discussion by TimWaeckerle** [Freaky2164/TowerDefense#5] [General]
Hi all!

Check out our new [SRS](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Software%20Requirement%20Specification/SRS.md).

Thanks for your contribution!

Tower Defense Team

**Comment by MyMrMush** [comment]
Hey there 👋 
I have taken a look at your SRS, following my feedback to some of the specifications:

3.1.1 Create Game:
I believe the verifiability could be a bit ambiguous here, as there are two different outcomes of clicking on create game - creating a game or resuming a saved game. Shouldn't the later be upon pressing load game?

3.1.2 Start Round:
Have you given thought on what should happen / how to prevent a user to press "new round" during a running round? This is not clear to me after reading this specification.

In general:
I really like your template for the specifications, especially the sections of pre- and post conditions. I believe this will make it easier to create test cases later on and makes the specifications verifiable.

Looking forward to reading more from you, good look with your implementation :)
Cheers,
Florian

**Reply by argastle** [reply]
Hi Florian,

We updated the UseCases of Create Game and Start Game. Load Game was not supposed to be in the UseCase of Create Game, it was an internal communication error. In Start Game, we updated the preconditions to secure the flow of events and prevent clicking the button during a running round. 

Thanks for reading our SRS carefully and we hope to hear from you again!

~ TowerDefense Team
